cirq/sim/act_on_state_vector_args.py

```diff
@@ -227,3 +228,33 @@ def _strat_act_on_state_vector_from_mixture(action: Any,
                                   out=args.available_buffer)
     args.swap_target_tensor_for(args.available_buffer)
     return True
+
+
+def _strat_act_on_state_vector_from_channel(action: Any,
+                                            args: 'cirq.ActOnStateVectorArgs'
+                                           ) -> bool:
+    kraus_operators = protocols.channel(action, default=None)
+    if kraus_operators is None:
+        return NotImplemented
+
+    shape = protocols.qid_shape(action)
+    kraus_tensors = [e.reshape(shape * 2).astype(args.target_tensor.dtype)
+                     for e in kraus_operators]
+    p = args.prng.random()
+    weight = None
+    for i in range(len(kraus_tensors)):
+        linalg.targeted_left_multiply(
+            left_matrix=kraus_tensors[i],
+            right_target=args.target_tensor,
+            target_axes=args.axes,
+            out=args.available_buffer,
+        )
+        weight = np.linalg.norm(args.available_buffer)**2
+        p -= weight
+        if p < 0:
+            break
+
```

Comment:
I would also assert p<0 after the loop, as a sanity check.

Reply:
It may not be the case that `p < 0`, due to floating point error, though it should be very small if not.